src/ray/core_worker/core_worker.cc

```diff
@@ -1065,6 +1072,24 @@ void CoreWorker::HandleGetObjectStatus(const rpc::GetObjectStatusRequest &reques
   }
 }
 
+void CoreWorker::HandleForceKillActor(const rpc::ForceKillActorRequest &request,
+                                      rpc::ForceKillActorReply *reply,
+                                      rpc::SendReplyCallback send_reply_callback) {
+  ActorID intended_actor_id = ActorID::FromBinary(request.intended_actor_id());
+  if (intended_actor_id != worker_context_.GetCurrentActorID()) {
+    std::ostringstream stream;
+    stream << "Mismatched ActorID: ignoring ForceKillActor for previous actor "
+           << intended_actor_id
+           << ", current actor ID: " << worker_context_.GetCurrentActorID();
+    auto msg = stream.str();
+    RAY_LOG(ERROR) << msg;
+    send_reply_callback(Status::Invalid(msg), nullptr, nullptr);
+    return;
+  }
+  RAY_LOG(INFO) << "Got ForceKillActor, shutting down...";
```

Comment:
can you log the actor ID too?

Reply:
Added another log message that logs the actorID on creation

Reply:
Sorry if this was unclear - I was more concerned about the situation where we couldn't trace which actor specifically was force killed. Does it make sense to log an ActorID with this "Shutting down" message?

Reply:
It's logged in the other log message and they both go to the logfile, not stderr.

Reply:
Thanks, got it.